clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -528,7 +528,6 @@ public void onFailure(RuntimeException e) {
                 }
 
                 private void recordRebalanceFailure() {
-                    state = MemberState.UNJOINED;
```

Comment:
This is a piggy-backed cleanup: we call resetGenerationXXX in the join/sync-group handler func and hence should not re-call it again here.

Reply:
Did you mean to say `recordRebalanceFailure` or is this comment just out of date after the latest changes?

Reply:
Previously this function has two lines: update the state and record sensors. Now that the first is called in the caller, this function becomes a one-liner and hence not worthy anymore so I in-lined it.